gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/password/ResetPasswordEndpoint.java

```diff
@@ -34,45 +33,46 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import static io.gravitee.am.gateway.handler.common.utils.ConstantKeys.TOKEN_PARAM_KEY;
-
 /**
  * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
  * @author GraviteeSource Team
  */
-public class ResetPasswordEndpoint implements Handler<RoutingContext> {
+public class ResetPasswordEndpoint extends AbstractEndpoint implements Handler<RoutingContext> {
 
     private static final Logger logger = LoggerFactory.getLogger(ResetPasswordEndpoint.class);
 
     private final ThymeleafTemplateEngine engine;
+    private final Domain domain;
 
-    public ResetPasswordEndpoint(ThymeleafTemplateEngine engine) {
+    public ResetPasswordEndpoint(ThymeleafTemplateEngine engine, Domain domain) {
         this.engine = engine;
+        this.domain = domain;
     }
 
     @Override
     public void handle(RoutingContext routingContext) {
-        final HttpServerRequest request = routingContext.request();
-        final String error = request.getParam(ConstantKeys.ERROR_PARAM_KEY);
-        final String errorDescription = request.getParam(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY);
-        final String success = request.getParam(ConstantKeys.SUCCESS_PARAM_KEY);
-        final String warning = request.getParam(ConstantKeys.WARNING_PARAM_KEY);
-        final String token = request.getParam(TOKEN_PARAM_KEY);
-        // add query params to context
+        HttpServerRequest request = routingContext.request();
+        // retrieve client (if exists)
+        Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
+        routingContext.put(ConstantKeys.PASSWORD_CONFIG_PARAM_KEY, PasswordSettings.getInstance(client, domain));
```

Comment:
Do not inject the Optional here but rather the value, this way we keep coherency with the others attributes.

Reply:
I forget that PasswordSettings.getInstance() return an optional, thank's